eodatasets3/scripts/tostac.py

```diff
@@ -19,55 +19,111 @@
-                # {
-                #     "rel": "self",
-                #     "href": '?',
-                # },
-                {"rel": "odc_product", "href": dataset.product.href},
-                {
-                    "rel": "alternative",
-                    "type": "text/html",
-                    "href": f"https://explorer.dea.ga.gov.au/dataset/{dataset.id}",
-                },
-            ],
-        )
-
         name = input_metadata.stem.replace(".odc-metadata", "")
         output_path = input_metadata.with_name(f"{name}.stac-item.json")
 
+        # Load STAC template data
+        if stac_template:
+            with open(stac_template, "r") as f:
+                stac_data = json.load(f)
+        else:
+            stac_data = {}
+
+        # Create STAC dict
+        item_doc = create_stac(
+            dataset, input_metadata, output_path, stac_data, base_url
+        )
+
         with output_path.open("w") as f:
             json.dump(item_doc, f, indent=4, default=json_fallback)
 
         echo(output_path)
 
 
+def create_stac(dataset, input_metadata, output_path, stac_data, base_url):
+    """
+    Create STAC document
+
+    :param dataset: Dict of the metadata content
+    :param input_metadata: Path of the Input metadata file
+    :param output_path: Path of the STAC output file
+    :param stac_data: Dict of the static STAC content
+    :param base_url: Base URL path for STAC file
+    :return: Dict of the STAC content
+    """
+
+    project = partial(
+        pyproj.transform, pyproj.Proj(init=dataset.crs), pyproj.Proj(init="epsg:4326"),
+    )
+    wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
+    item_doc = dict(
+        stac_version=stac_data["stac_version"]
+        if "stac_version" in stac_data
+        else "1.0.0-beta.1",
+        stac_extensions=stac_data["stac_extensions"]
+        if "stac_extensions" in stac_data
+        else [],
+        id=dataset.id,
+        type=stac_data["type"] if "type" in stac_data else "Feature",
+        bbox=wgs84_geometry.bounds,
+        geometry=wgs84_geometry.__geo_interface__,
+        properties={**dataset.properties, "odc:product": dataset.product.name},
+        # TODO: Currently assuming no name collisions.
+        assets={
+            **{
+                name: (
+                    {**stac_data["assets"][name], "href": base_url + m.path}
+                    if "assets" in stac_data and name in stac_data["assets"]
+                    else {"href": base_url + m.path}
+                )
+                for name, m in dataset.measurements.items()
+            },
+            **{
+                name: (
+                    {**stac_data["assets"][name], "href": base_url + m.path}
+                    if "assets" in stac_data and name in stac_data["assets"]
+                    else {"href": base_url + m.path}
+                )
+                for name, m in dataset.accessories.items()
+            },
+        },
+        links=[
```

Comment:
Do you think we should add a `parent` link to the source document, maybe on THREDDS?